### PR TITLE
feat(faq): add schema, add subheading formatting

### PR DESCRIPTION
**Pull request by qiyundai**
Please always provide the [JIRA issue(s)](https://jira.corp.adobe.com/secure/RapidBoard.jspa?rapidView=34618) your PR is for, as well as test URLs where your change can be observed (before and after):

Feat [MWPW-116568](https://jira.corp.adobe.com/browse/MWPW-116568) & Fix [MWPW-116215](https://jira.corp.adobe.com/browse/MWPW-116215)

Test URLs:
- Before: https://main--express-website--adobe.hlx.page/express/create/logo
- After: https://mwpw-116568-116215--express-website--webistry-development.hlx3.page/express/create/logo

### CODE REVIEW

File: express/blocks/faq/faq.js [hunk 1]
```diff
@@ -35,15 +67,23 @@ function decorateFAQBlocks($block) {
...
+
+    appendToSchema($schemaScript, question, answer);
   });
 
+  if (showSchema !== 'no') {
+    document.head.appendChild($schemaScript);
+  } else {
+    $schemaScript.remove();
+  }
+
```
Comment (rofe) [inline express/blocks/faq/faq.js:86]: Do authors really need to be able to turn the schema off?

**Review by rofe — CHANGES_REQUESTED** [review]
@qiyundai LGTM but... since both `initSchema` and `appendToSchema` are both only used inside `decorateFAQBlocks`, I wonder if we could simplify the implementation by merging it into a single `addSchema(entities)` which gets called at the end of the block decoration?

File: express/blocks/faq/faq.js (hunk 1 above)
Comment (qiyundai) [reply express/blocks/faq/faq.js:86]: That's what they said in the Jira issue. I wonder the same. We can maybe bring that up to PMs later today?

**Comment by qiyundai** [comment]
> @qiyundai LGTM but... since both `initSchema` and `appendToSchema` are both only used inside `decorateFAQBlocks`, I wonder if we could simplify the implementation by merging it into a single `addSchema(entities)` which gets called at the end of the block decoration?

Sure, we can do that! I had this debate while writing it as well. I'm probably missing the point but here's the reasoning for the current approach -- I wanted to leverage the existing foreach loop over the 'faqs', which is in the collapsed range of code under the files changed tab.

File: express/blocks/faq/faq.js (hunk 1 above)
Comment (rofe) [reply express/blocks/faq/faq.js:86]: Yes. I think it's adding unnecesary complexity if not really needed.

**Comment by rofe** [comment]
> > @qiyundai LGTM but... since both `initSchema` and `appendToSchema` are both only used inside `decorateFAQBlocks`, I wonder if we could simplify the implementation by merging it into a single `addSchema(entities)` which gets called at the end of the block decoration?
> 
> Sure, we can do that! I had this debate while writing it as well. I'm probably missing the point but here's the reasoning for the current approach -- I wanted to leverage the existing foreach loop over the 'faqs', which is in the collapsed range of code under the files changed tab.

Well, it may well be me who's missing something :) 

You should be able to leverage the same loop, just instead of adding to the schema, you add to an array of `entities`. And at the end you write the entire schema.

**Comment by qiyundai** [comment]
> > > @qiyundai LGTM but... since both `initSchema` and `appendToSchema` are both only used inside `decorateFAQBlocks`, I wonder if we could simplify the implementation by merging it into a single `addSchema(entities)` which gets called at the end of the block decoration?
> > 
> > 
> > Sure, we can do that! I had this debate while writing it as well. I'm probably missing the point but here's the reasoning for the current approach -- I wanted to leverage the existing foreach loop over the 'faqs', which is in the collapsed range of code under the files changed tab.
> 
> Well, it may well be me who's missing something :)
> 
> You should be able to leverage the same loop, just instead of adding to the schema, you add to an array of `entities`. And at the end you write the entire schema.

Done! It does look&&feel much prettier.

**Review by rofe — APPROVED** [review]

**Pull request merged**